app/models/claims/calculations.rb

```diff
@@ -2,9 +2,15 @@ module Claims::Calculations
   def calculate_fees_total(category = nil)
     fees.reload
     if category.blank?
-      fees.map(&:amount).compact.sum
+      fees.map do |fee|
```

Comment:
This piece of code is far from the ideal, but the whole fee amount calculation is a bit convoluted, so this was the only way I managed to do to calculate the fee amount:

- it tries to calculate the fee amount (there's cases where it does not calculate, but when it does, the method also updates the amount attribute. Arguably it should potentially just calculate an amount and that amount would have to be assigned out it, but I left it unchanged out of fear of affecting other parts of the process :S )
- it always returns the fee amount, mainly because there's times where the calculate_amount does not execute, but the fee amount itself is already set.

This should be revisited to try to simplify the convoluted way the way it works atm.

Reply:
I'd create an issue or put a `#TODO/FIXME` in the code to cover this